src/ansys/geometry/core/misc/checks.py

```diff
@@ -302,3 +303,44 @@ def check_type_all_elements_in_iterable(
     """
     for elem in input:
         check_type(elem, expected_type)
+
+
+def min_backend_version(method_version: str):
+    """
+    Compare the min version required by a method to the current backend version.
+
+    Parameters
+    ----------
+    method_version : str
+        The minimum version required by the method.
+
+    Raises
+    ------
+    GeometryRuntimeError
+        If the method version is higher than the backend version.
+    GeometryRuntimeError
+        If the client is not available.
+    """
+    # Lazy import to avoid circular imports
+    from ansys.geometry.core.errors import GeometryRuntimeError
+
+    def backend_version_decorator(method):
+        def wrapper(self, *args, **kwargs):
+            if self._grpc_client is None:
+                raise GeometryRuntimeError(
```

Comment:
I'm unsure whether there is any scenario where the client is not defined (as long as the attribute exists). We always pass it in construction to all objects. But in any case, good "defensive" measure